src/Helpers/Cache/FileSystemCacheHandler.php

```diff
@@ -19,8 +19,11 @@ class FileSystemCacheHandler implements CacheHandler
     public function __construct($temp_directory_prefix = 'auth0-php')
     {
         $this->tmp_dir = sys_get_temp_dir().DIRECTORY_SEPARATOR.$temp_directory_prefix.DIRECTORY_SEPARATOR;
-        if (! file_exists($this->tmp_dir)) {
-            mkdir($this->tmp_dir);
+        if (! is_dir($this->tmp_dir)
+            && ! @mkdir($this->tmp_dir, 0777, true)
```

Comment:
```suggestion
            && ! mkdir($this->tmp_dir, 0777, true)
```

I am not wild about using `@` for any reason. If this PR is fixing the race condition, why would we need to suppress the error?

Reply:
Because in some rare conditions like I had you would have the php warning while everything is ok. This warning is replaced by the exception that fails the app explicitly if it really fails creating the cache directory which unlikely to happen if current user permissions are OK and the disk is not full.

Reply:
How often is it trying to create this directory? Does this temporary directory change regularly?

Reply:
I guess it depends on the machine where the code lives. In my case it changes at least on each deployment.